src/business-logic/services/database-change-ingest-service.ts

```diff
@@ -40,46 +55,170 @@ export class DatabaseChangeIngestService implements IngestChangeService {
-    segmentChangedListener.onDeleted(segmentId => this.enqueueEvent(() => this.deleteSegment(segmentId)))
+  private async syncDataWithIngestedData(): Promise<void> {
+    const ingestedRundowns: IngestedRundown[] = await this.ingestedRundownRepository.getIngestedRundowns()
+    await this.deleteRundownsNotPresentInIngestedRundowns(ingestedRundowns)
+
+    await Promise.all(ingestedRundowns.map(async (ingestedRundown) => {
+      let oldRundown: Rundown | undefined
+      try {
+        oldRundown = await this.rundownRepository.getRundown(ingestedRundown.id)
+      } catch (exception) {
+        if (!(exception instanceof NotFoundException)) {
+          throw exception
+        }
+        // The Rundown doesn't exist in the database which means it's a brand new Rundown.
+      }
+
+      // If the Rundown isn't active, we can simply just "re-ingest" it into our database collection as a fresh Rundown.
+      const updatedRundown: Rundown = oldRundown && oldRundown.isActive()
+        ? this.updateActiveRundownFromIngestedRundown(ingestedRundown, oldRundown)
+        : this.createNewRundownFromIngestedRundown(ingestedRundown)
+
+      await this.rundownRepository.deleteRundown(ingestedRundown.id) // Delete the old Rundown to get rid of deleted Entities
+      await this.rundownRepository.saveRundown(updatedRundown) // Save the new Rundown
+      this.eventEmitter.emitRundownUpdated(updatedRundown)
+    }))
+  }
+
+  private async deleteRundownsNotPresentInIngestedRundowns(ingestedRundowns: IngestedRundown[]): Promise<void> {
+    const ingestedRundownsIds: string[] = ingestedRundowns.map(ingestedRundown => ingestedRundown.id)
+    const basicRundowns: BasicRundown[] = await this.rundownRepository.getBasicRundowns()
+    const rundownIds: string[] = basicRundowns.map(basicRundown => basicRundown.id)
+
+    for (const rundownId of rundownIds) {
+      // Delete all Rundowns that has been deleted and therefore no longer exists in the IngestedRundowns.
+      if (!ingestedRundownsIds.includes(rundownId)) {
+        await this.rundownRepository.deleteRundown(rundownId)
+        this.eventEmitter.emitRundownDeleted(rundownId)
+      }
+    }
+  }
+
+  private updateActiveRundownFromIngestedRundown(ingestedRundown: IngestedRundown, rundown: Rundown): Rundown {
+    const ingestedSegmentIds: string[] = ingestedRundown.ingestedSegments.map(ingestedSegment => ingestedSegment.id)
+    const segmentsToBeDeleted: Segment[] = rundown.getSegments().filter(segment => !ingestedSegmentIds.includes(segment.id))
+    segmentsToBeDeleted.forEach(segment => rundown.removeSegment(segment.id))
+
+
+    ingestedRundown.ingestedSegments.forEach(ingestedSegment => {
+      const segmentOnRundown: Segment | undefined = rundown.getSegments().find(segment => segment.id === ingestedSegment.id)
+      if (!segmentOnRundown) {
+        const newSegment: Segment = this.ingestedEntityToEntityMapper.fromIngestedSegment(ingestedSegment)
```

Comment:
Is `new` necessary?

Reply:
Technically no, but I added it because I wanted to emphasize that we are creating a brand new Segment here.